app/controllers/album.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 const logger = require('../logger'),
-  requests = require('../services/albums'),
+  Album = require('../models').album,
   errors = require('../errors'),
-  rp = require('request-promise');
+  requests = require('../services/albums');
 
 exports.getAlbums = (req, res, next) => {
   requests
     .getAlbums()
     .then(response => {
       res.send(response);
     })
-    .catch(err => next(errors.fetchError(err.message)));
+    .catch(next);
+};
+
+exports.purchaseAlbum = (req, res, next) => {
+  requests
+    .getAlbums()
+    .then(response => {
+      const albumId = Number(req.params.id);
+      if (!albumId) next(errors.albumNotFound('Missing album ID.'));
+      const album = response.find(element => {
+        return element.id === albumId;
+      });
+      if (album) {
+        Album.FOCAlbum({
+          id: album.id,
+          artistId: album.userId,
+          title: album.title
+        })
+          .spread((dbAlbum, created) => {
+            dbAlbum.getUsers({ where: { email: req.user.email } }).then(owner => {
+              if (owner.length) {
```

Comment:
Check buy duplicate album with a constraint in DB (unique by user id and album id)